muze/src/app/api/user/route.ts

```
@@ -0,0 +1,126 @@
+'use server';
+import { CreateUser, UpdateUser, UploadPhoto } from '@/db/UserUpdate';
+import { GetUserById, GetUsersByUsername } from '@/db/UserGet';
+import { checkSession } from '@/utils/serverSession';
+
+// Create a user
+export async function createUser(token: {
+    access_token: string | undefined;
+    token_type: string | undefined;
+    expires_at: number;
+    expires_in: number;
+    refresh_token: string | undefined;
+    scope: string | undefined;
+    id: string;
+    name?: string | null;
+    email?: string | null;
+    picture?: string | null;
+    sub?: string;
+}) {
+    try {
+        await CreateUser(token);
+    } catch (error) {
+        console.error(error);
+        return null;
+    }
+    return true;
+}
+
+export async function updateUser(
+    username: string,
+    bio: string,
+    file: File | null,
+    profile_pic?: string | null
+) {
+    // Username validation
+    if (username.length < 2 ) {
+        console.error('Username is too short. Username must be at least 2 characters');
+        return null;
```

will note that returning null from all of these states doesn't give us a good way to distinguish different failure modes. for ex, if this updateUser function fails, it would be hard to tell whether it's from the username being invalid or the profile picture being invalid. we should consider either refactoring into separate updateUsername and updateProfile picture functions or refactoring our error handling so we can propagate the error more effectively. 

also worth nothing that in its current form, if the username was invalid, the profile picture would not upload, which doesn't seem like a very intuitive experience. think this is fine for now, but will have to put some thought into the user experience later down the line.

---

I think it makes sense that if one part of your profile isn't updated correctly, the entirety of your profile shouldn't change. (i.e. no remote upload should occur).

This is so that we can reduce API calls; we can add frontend validation checks.